source/Nuke.Common/Tools/OctoVersion/OctoVersionAttribute.cs

```diff
@@ -22,27 +24,232 @@ namespace Nuke.Common.Tools.OctoVersion
     [UsedImplicitly(ImplicitUseKindFlags.Default)]
     public class OctoVersionAttribute : ValueInjectionAttributeBase
     {
+        /// <summary>
+        /// Which framework to use when selecting the OctoVersion library from the package
+        /// </summary>
         public string Framework { get; set; } = "net5.0";
-        public bool UpdateBuildNumber { get; set; }
+
+        /// <summary>
+        /// Whether to update the build number and output octoversion attributes into the CI platform
+        /// Supports AzurePipelines, TeamCity, AppVeyor and GitHubActions
+        /// </summary>
+        public bool UpdateBuildNumber { get; set; } = NukeBuild.IsServerBuild;
```

Comment:
This is dead-code I believe, because inside the if-branch it's using the null-shorthand like `<Host>.Instance?...`.

Reply:
By dead code are you referring to having the property at all?  Or having the default value set to `NukeBuild.IsServerBuild` ?

FYI we are using this parameter to set whether to autodetect the CI platform (which causes OctoVersion to emit variables into the CI platform... unrelated to Nuke specifically)... and also using the parameter in the Nuke attribute to specifically SetBuildNumber on the 3 CI platforms Nuke supports

Reply:
Sorry, I just meant the default value. It can just be `true` because anyway on local execution there is nothing to execute it on. Maybe though we could set `Console.Title`, but that's a different topic.

Reply:
that parameter is also being used to tell octoversion tool whether to auto detect environment or not.  However due to some other reasons ive decided to default to just `true` anyhow 👍